providers/openshift.md

```diff
@@ -32,9 +41,146 @@ master_hostname
+
+
+#### Quickstart
+
+Since VirtualBox is a consistent hypervisor across all platforms, this guide
+will assume that is being used, and updates to some commands might be
+necessary.  Most commands should be virtualization software agnostic and work
+regardless of the hypervisor.
 
-openshift_use_manageiq is set to True to configure the Service Account needed by ManageIQ. there are more like variables like use_metrics to enable other options in OpenShift.
-A more detailed example will be found in inventory/byo/hosts.example.
-```ansible-playbook playbooks/byo/config.yml -i path/to/inventory/file```
+1. Create an addon directory for manageiq in minishift.  Something like:
+  
+  ```console
+  $ mkdir -p ./minifshift/addons/manageiq
+  ```
+  
+2. Create a `manageiq.addon` file with the following contents in that dir:
+  
+  ```bash
+  # Name: manageiq
+  # Description:  Basic Users, Roles, and Policies necessary to have ManageIQ/CFME connect to Openshift
+  
+  oc adm new-project management-infra --description="Management Infrastructure"
+  
+  oc create serviceaccount management-admin -n 'management-infra'
+  oc create serviceaccount inspector-admin -n 'management-infra'
+  
+  oc create -f management-infra-admin-cluster-role.json
+  oc create -f hawkular-metrics-admin-cluster-role.json
+  
+  oc policy add-role-to-user -n management-infra admin -z management-admin
+  oc policy add-role-to-user -n management-infra management-infra-admin -z management-admin
+  oc policy add-cluster-role-to-user cluster-reader system:serviceaccount:management-infra:management-admin
+  oc policy add-scc-to-user privileged system:serviceaccount:management-infra:management-admin
+  oc policy add-cluster-role-to-user system:image-puller system:serviceaccount:management-infra:inspector-admin
+  oc policy add-scc-to-user privileged system:serviceaccount:management-infra:inspector-admin
+  oc policy add-cluster-role-to-user self-provisioner system:serviceaccount:management-infra:management-admin
+  oc policy add-cluster-role-to-user hawkular-metrics-admin system:serviceaccount:management-infra:management-admin
+  ```
+  
+3. Create a management-infra-admin-cluster-role.json file with the following
+   contents in that dir:
+  
+  ```json
+  {
+    "apiVersion": "v1",
+    "kind":       "ClusterRole",
+    "metadata": {
+      "name": "management-infra-admin"
+    },
+    "rules": [
+      {"resources": ["pods/proxy"],"verbs":["*"]}
+    ]
+  }
+  ```
+  
+3. Create a hawkular-metrics-admin-cluster-role.json file with the following
+   contents in that dir:
+  
+  ```json
+  {
+    "apiVersion": "v1",
+    "kind":       "ClusterRole",
+    "metadata": {
+      "name": "hawkular-metrics-admin"
+    },
+    "rules": [
+      {
+        "apiGroups": [""],
+        "resources": ["hawkular-metrics","hawkular-alerts"],
+        "verbs":     ["*"]
+      }
+    ]
+  }
+  ```
```

Comment:
Probably not, but I was mostly just extracting/reverse engineering exactly what I saw in the Ansible playbook into a script since I probably spent a good day trying to figure out a proper permission configuration to get `manageiq` a token to do a refresh...

Might be nice though to have an addition to this setup (at somepoint) to deploy hawkular instance, and connect it so that developers can mess with that as well.  Probably will do that in a separate PR.